2025/shell_scripting/backup_with_rotation.sh

```
@@ -0,0 +1,33 @@
+
+# Create the backup zip
+sudo zip -r "$backup_dir/$file_nm" "$dest"
+
+# Change to backup directory
+cd "$backup_dir" || exit
+
+# Initialize counter
+count=0
+
+# Loop through sorted files and delete older backups
+for file in $(ls -t); do
```

Shellcheck (0.10.0)

[warning] Iterating over ls output is fragile. Use globs.

(SC2045)